### PR TITLE
Fixed bugs with mix node implementations and usage.

### PR DESCRIPTION
**Pull request by niklasharrysson**
- Fixed bugs with mix nodes.
- Fixed bugs in standard_surface graph.
- Fixed errors in standard_surface examples.

### CODE REVIEW

**Comment by niklasharrysson** [comment]
Main issue was inconsistencies in implementation and usage of mix nodes. MaterialX's mix has the opposite convention as OSL and GLSL mix in terms of the order of the inputs given och and how the mix weight is applied to them.

File: resources/Materials/Examples/StandardSurface/standard_surface_brass_tiled.mtlx [hunk 1]
```diff
@@ -16,12 +16,11 @@
   <material name="Tiled_Brass">
     <shaderref name="SR_brass1" node="standard_surface">
       <bindinput name="base" type="float" value="1" />
+      <bindinput name="base_color" type="color3" nodegraph="NG_brass1" output="out_color" />
       <bindinput name="specular" type="float" value="0" />
+      <bindinput name="specular_color" type="color3" value="1.0, 1.0, 1.0" />
       <bindinput name="specular_roughness" type="float" nodegraph="NG_brass1" output="out_roughness" />
       <bindinput name="metalness" type="float" value="1" />
-      <bindinput name="coat" type="float" value="1" />
```
Comment (niklasharrysson) [inline resources/Materials/Examples/StandardSurface/standard_surface_brass_tiled.mtlx]: Fix up parameters for a metal. Coat is not meant to be used for this.

File: libraries/bxdf/standard_surface.mtlx [hunk 2]
```diff
@@ -64,9 +64,9 @@
       <input name="in2" type="float" interfacename="coat_roughness" />
     </multiply>
     <mix name="coat_affected_rougness" type="float">
-      <input name="fg" type="float" interfacename="specular_roughness" />
-      <input name="bg" type="float" nodename="coat_affect_roughness_multiply2" />
-      <input name="mix" type="float" value="1.0" />
+      <input name="fg" type="float" value="1.0" />
```
Comment (bernardkwok) [inline libraries/bxdf/standard_surface.mtlx]: Just checking here as you swapped all 3 terms. I assume that it correct now and the mixing comes from upstream.

File: resources/Materials/Examples/StandardSurface/standard_surface_brass_tiled.mtlx (hunk 1 above)
Comment (jstone-lucasfilm) [reply resources/Materials/Examples/StandardSurface/standard_surface_brass_tiled.mtlx]: If I'm remembering correctly, this unusual usage of "coat over metal" to represent brass was borrowed from one of the metals in the public Standard Surface library for Maya.  Not that we need to follow this convention in our material examples, but I believe that's the origin.

**Comment by jstone-lucasfilm** [comment]
> Main issue was inconsistencies in implementation and usage of mix nodes. MaterialX's mix has the opposite convention as OSL and GLSL mix in terms of the order of the inputs given och and how the mix weight is applied to them.

Because MaterialX uses the fg/bg convention instead of in1/in2, it's really just the _presentation_ of the inputs that's reversed, and we might consider simply switching this presentation order, so that "bg" comes before "fg" in the specification and node declaration.  This shouldn't have any impact on any existing materials, since inputs are always referenced explicitly by their name.

**Comment by niklasharrysson** [comment]
> > Main issue was inconsistencies in implementation and usage of mix nodes. MaterialX's mix has the opposite convention as OSL and GLSL mix in terms of the order of the inputs given och and how the mix weight is applied to them.
> 
> Because MaterialX uses the fg/bg convention instead of in1/in2, it's really just the _presentation_ of the inputs that's reversed, and we might consider simply switching this presentation order, so that "bg" comes before "fg" in the specification and node declaration. This shouldn't have any impact on any existing materials, since inputs are always referenced explicitly by their name.

That sounds like a very good idea. For artists connecting nodes in a graph maybe it doesn't matter much, if they are familiar with foreground/background compositing, but as a programmer I find it confusing as I think of the input ordering as function arguments :)

File: libraries/bxdf/standard_surface.mtlx (hunk 2 above)
Comment (niklasharrysson) [reply libraries/bxdf/standard_surface.mtlx]: Yes, this is the equation for modulating roughness in https://autodesk.github.io/standard-surface/ (section 3.2)

**Pull request merged**